pytest_catchlog.py

```diff
@@ -92,19 +97,80 @@ def pytest_addoption(parser):
     add_option_ini(parser,
                    '--log-format',
-                   dest='log_format', default=('%(filename)-25s %(lineno)4d'
-                                               ' %(levelname)-8s %(message)s'),
+                   dest='log_format',
+                   default=DEFAULT_LOG_FORMAT,
                    help='log format as used by the logging module.')
     add_option_ini(parser,
                    '--log-date-format',
-                   dest='log_date_format', default=None,
+                   dest='log_date_format',
+                   default=DEFAULT_DATE_FORMAT,
                    help='log date format as used by the logging module.')
+    add_option_ini(parser,
+                   '--log-level-extra',
+                   dest='log_level_extra',
+                   type='args',
+                   action='append',
+                   help='extra logging levels that verbosity should be sensible to.')
 
 
 def pytest_configure(config):
     """Always register the log catcher plugin with py.test or tests can't
     find the  fixture function.
     """
+    # Prepare the available levels dictionary
+    available_levels = set([
+        logging.CRITICAL,
+        logging.ERROR,
+        logging.WARNING,
+        logging.INFO,
+        logging.DEBUG,
+        logging.NOTSET
+    ])
+    for level in get_option_ini(config, 'log_level_extra'):
+        try:
+            level_num = int(getattr(logging, level, level))
+        except ValueError:
+            # Python logging does not recognise this as a logging level
+            raise pytest.UsageError(
+                "'{0}' is not recognized as a logging level name. Please "
+                "consider passing the logging level num instead.".format(
+                    level))
+        else:
+            if not (logging.NOTSET <= level_num <= logging.CRITICAL):
+                raise pytest.UsageError(
+                    "'{0}' is ignored as not being in the valid logging "
+                    "levels range: NOTSET({1}) - CRITICAL({2})".format(
+                        level,
+                        logging.NOTSET,
+                        logging.CRITICAL))
+
+        available_levels.add(level_num)
+
+    # Build a dictionary mapping of verbosity level to logging level, as
+    # follows, unless there's custom levels added
+    # verbosity=0         CRITICAL (no pytest output is shown and CRITICAL
+    #                               log messages are displayd)
+    # verbosity=1   -v    ERROR    (pytest verbosity kicks in, but only
+    #                               log messages with higher or equal
+    #                               level to ERROR are shown)
+    # verbosity=2   -vv   WARNING  (start showing log messages with a
+    #                               higher or equal level to WARNING)
+    # verbosity=4   -vvv  INFO
+    # - ... etc
+    handled_levels = dict(enumerate(sorted(available_levels, reverse=True)))
```

Comment:
I haven't followed the previous discussion, but intuitively to me it'd make sense to show `CRITICAL` and `ERROR` without `-v` (maybe even `WARNING`), and then have `-v` or `-vv` for `INFO`. Not displaying errors without `-v` sounds wrong, IMHO.